src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -494,6 +496,69 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
             Assert.True(fastestLength >= optimalLength);
             Assert.True(optimalLength >= smallestLength);
         }
+
+        [Theory]
+        [MemberData(nameof(ZLibOptionsRoundTripTestData))]
+        public async Task RoundTripWithZLibCompressionOptions(string testFile, ZLibCompressionOptions options)
+        {
+            using var uncompressedStream = await LocalMemoryStream.readAppFileAsync(testFile);
+            var compressedStream = CompressTestFile(uncompressedStream, options);
+            using var decompressor = CreateStream(compressedStream, mode: CompressionMode.Decompress);
+            using var decompressorOutput = new MemoryStream();
+            await decompressor.CopyToAsync(decompressorOutput);
+            decompressor.Dispose();
+            decompressorOutput.Position = 0;
+            uncompressedStream.Position = 0;
+
+            byte[] uncompressedStreamBytes = uncompressedStream.ToArray();
+            byte[] decompressorOutputBytes = decompressorOutput.ToArray();
+
+            Assert.Equal(uncompressedStreamBytes.Length, decompressorOutputBytes.Length);
+            for (int i = 0; i < uncompressedStreamBytes.Length; i++)
+            {
+                Assert.Equal(uncompressedStreamBytes[i], decompressorOutputBytes[i]);
+            }
+        }
+
+        private MemoryStream CompressTestFile(LocalMemoryStream testStream, ZLibCompressionOptions options)
+        {
+            var compressorOutput = new MemoryStream();
+            using (var compressionStream = CreateStream(compressorOutput, options, leaveOpen: true))
+            {
+                var buffer = new byte[4096];
+                int bytesRead;
+                while ((bytesRead = testStream.Read(buffer, 0, buffer.Length)) > 0)
+                {
+                    compressionStream.Write(buffer, 0, bytesRead);
+                }
+            }
+
+            compressorOutput.Position = 0;
+            return compressorOutput;
+        }
+
+        protected async void CompressionLevel_SizeInOrderBase(string testFile)
```

Comment:
From https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void :
```suggestion
        protected async Task CompressionLevel_SizeInOrderBase(string testFile)
```